### PR TITLE
variable: Add getters for global/session scope

### PR DESCRIPTION
**Pull request by morgo**
### What problem does this PR solve?

Problem Summary:

Previously special instance scoped variables used the function `GetSessionOnlySysVars` with a large switch statement. There are setter functions on the sysvar struct, so it seems natural to move these special gets to getters instead.

Ideally there can be getters used for some of the special handling in session/session.go for tikv vars. This will require some additional work since a sessionctx (not sessionvars) is required to be passed to run internal sql.

### What is changed and how it works?

What's Changed:

Internal refactoring: move `GetSessionOnlySysVars` to getters on the sysvar struct.

### Related changes

- None

### Check List 

Tests 

- Covered by existing unit tests

Side effects

- More complicated cherry picks

### Release note 

- No release note

### CODE REVIEW

File: sessionctx/variable/sysvar.go [hunk 1]
```diff
@@ -119,12 +120,47 @@ type SysVar struct {
...
+	if sv.GetSession != nil {
+		return sv.GetSession(s)
+	}
+	if val, ok := s.stmtVars[sv.Name]; ok {
+		return val, nil
+	}
+	if val, ok := s.systems[sv.Name]; !ok {
+		return val, errors.New("sysvar has not yet loaded")
+	}
+	return s.systems[sv.Name], nil
```
Comment (xhebox) [inline sessionctx/variable/sysvar.go:161]: You could just return `val` instead of accessing the map again.

**Review by xhebox — COMMENTED** [review]
Rest LGTM

**Review by xiongjiwei — APPROVED** [review]
lgtm

**Pull request merged**